dev-tools/xtask/src/a4x2_package.rs

```diff
@@ -0,0 +1,382 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Subcommand: cargo xtask a4x2-package
+
+use anyhow::Result;
+use camino::{Utf8Path, Utf8PathBuf};
+use clap::Parser;
+use serde::Deserialize;
+use sha2::Digest;
+use std::env;
+use std::fs;
+use std::io::{Read, Write};
+use walkdir::WalkDir;
+use xshell::{cmd, Shell};
+
+#[derive(Parser)]
+pub struct A4x2PackageArgs {
+    /// Bundle omicron live tests into the output tarball
+    #[clap(long)]
+    live_tests: bool,
+
+    /// Bundle omicron end-to-end-tests package into the output tarball
+    #[clap(long)]
+    end_to_end_tests: bool,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "kebab-case")]
+struct NextestConfig {
+    nextest_version: NextestConfigVersion,
+}
+
+#[derive(Deserialize, Debug)]
+struct NextestConfigVersion {
+    // required: String,
+    recommended: String,
+}
+
+struct Environment {
+    cargo: String,
+    home_dir: Utf8PathBuf,
+    work_dir: Utf8PathBuf,
+    src_dir: Utf8PathBuf,
+    out_dir: Utf8PathBuf,
+    omicron_dir: Utf8PathBuf,
+    in_ci: bool,
+}
+
+pub fn run_cmd(args: A4x2PackageArgs) -> Result<()> {
+    let sh = Shell::new()?;
+
+    let env = {
+        // Buildomat sets the CI environment variable. We'll work a bit
+        // differently in that case (no cleanup, use /work)
+        let in_ci = env::var("CI").is_ok();
+
+        let cargo =
+            std::env::var("CARGO").unwrap_or_else(|_| String::from("cargo"));
+
+        let home_dir = Utf8PathBuf::from(std::env::var("HOME")?);
+        let work_dir = if in_ci {
+            Utf8PathBuf::from("/work/a4x2-package")
+        } else {
+            home_dir.join(".cache/a4x2-package")
+        };
+
+        let src_dir = Utf8PathBuf::try_from(env::current_dir()?)?;
+
+        // Delete any results from previous runs. We don't mind if there's errors.
+        fs::remove_dir_all(&work_dir).ok();
```

Comment:
Worth a confirmation here?  Or: would it make sense to create a new directory in `$work_dir` for this invocation so that we don't have to worry about colliding with a previous one?